homeassistant/components/renson/__init__.py

```diff
@@ -0,0 +1,181 @@
+        """Sync time of device."""
+        await hass.async_add_executor_job(renson_api.sync_time)
+
+    async def set_manual_level(call: ServiceCall) -> None:
+        """Set manual level."""
+        level_string = call.data.get("manual_level", "Off")
+        level = Level[level_string.upper()]
+
+        await hass.async_add_executor_job(renson_api.set_manual_level, level)
+
+    async def set_breeze(call: ServiceCall) -> None:
+        """Configure breeze feature."""
+        level = call.data.get("breeze_level", "")
+        temperature = call.data.get("temperature", 0)
+        activated = call.data.get("activate", False)
+
+        await hass.async_add_executor_job(
+            renson_api.set_breeze, level, temperature, activated
+        )
+
+    async def set_day_night_time(call: ServiceCall) -> None:
+        """Configure day night times."""
+        day = call.data.get("day", "7:00")
+        night = call.data.get("night", "22:00")
+
+        await hass.async_add_executor_job(renson_api.set_time, day, night)
+
+    async def set_pollution_settings(call: ServiceCall) -> None:
+        """Configure pollutions settings."""
+        day = call.data.get("day_pollution_level", "")
+        night = call.data.get("night_pollution_level", "")
+        humidity_control = call.data.get("humidity_control", "")
+        airquality_control = call.data.get("airquality_control", "")
+        co2_control = call.data.get("co2_control", "")
+        co2_threshold = call.data.get("co2_threshold", 0)
+        co2_hysteresis = call.data.get("co2_hysteresis", 0)
+
+        await renson_api.set_pollution(
+            day,
+            night,
+            humidity_control,
+            airquality_control,
+            co2_control,
+            co2_threshold,
+            co2_hysteresis,
+        )
+
+    async def set_filter_days(call: ServiceCall) -> None:
+        """Set filter dayes."""
+        days = call.data.get("days", 90)
+
+        await hass.async_add_executor_job(renson_api.set_filter_days, days)
+
+    hass.services.async_register(DOMAIN, "set_manual_level", set_manual_level)
+    hass.services.async_register(DOMAIN, "set_breeze", set_breeze)
+    hass.services.async_register(DOMAIN, "set_day_night_time", set_day_night_time)
+    hass.services.async_register(
+        DOMAIN, "set_pollution_settings", set_pollution_settings
+    )
+    hass.services.async_register(DOMAIN, "set_filter_days", set_filter_days)
+    hass.services.async_register(DOMAIN, "set_timer_level", set_timer_level)
+    hass.services.async_register(DOMAIN, "sync_time", sync_time)
+
+
+class RensonCoordinator(DataUpdateCoordinator):
+    """Data update coordinator for Renson."""
+
+    def __init__(self, hass, api, update_interval=timedelta(seconds=30)):
+        """Initialize my coordinator."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            # Name of the data. For logging purposes.
+            name="Renson",
```

Comment:
This name is now used for all Renson coordinators and RensonFirmware coordinators. Each coordinator should have a unique name.